### ISSUE TITLE
Get resolved instance of Promise<Void>

### ISSUE THREAD

**Issue by opfeffer** [mxcl/PromiseKit#532]
* PromiseKit version: 4.0.0
* Xcode version: Xcode8 GM
* Installation method:
  - [x] Carthage
  
I may be missing the obvious-- but how can I create a resolved instance of `Promise<Void>` ?

I know there is `public func asVoid()` but that assumes you have a Promise. What if I simply want a already resolved Promise without value?

`required public init(value: T)` doesn't work and `Promise()` doesn't appear to be working either.

Thanks for your help. Sorry for the possibly silly question!

**Comment by mxcl** [comment]
`Promise(value: ())`

**Comment by opfeffer** [comment]
#love.  thanks sir

**Comment by mxcl** [comment]
Also, it's not a stupid question, I was initially confused about how to do this too, and I wrote the code. `Promise()` used to work, but we elected to be more “Swifty” with v4 and thus the `value` parameter  is now mandatory, which means the `Void` initializer ends up being odd.

**Comment by evgeniyd** [comment]
OMG, I come up with the following code, before found this:

    let voidValue: Void = {return}();
    return Promise<Void>(value: voidValue)

KISS!

**Comment by ldiqual** [comment]
@mxcl I return void promises a lot in my code and I find both syntaxes (`Promise(value: Void())` and `Promise(value: ())`) to be confusing. I ended up adding `Promise.void`. Would it be appropriate to have that in the library? I'm thinking about something like this:

```
extension Promise {
    static var void: Promise<Void> {
        return Promise<Void>(value: ())
    }
}
```

And how I use it:

```
func doSomething() -> Promise<Void> {
    return .void
}
```

**Comment by mxcl** [comment]
Hmm I tried something like this:

```swift
extension Promise {
    static var void: Promise<Void> {
        return Promise<Void>(value: ())
    }
}
```

And Swift wouldn't compile it due to it requiring `T` to be inferred. I must have been doing it wrong.

Could you PR it?

**Comment by raygun101** [comment]
How about creating an extension for only <Void> Promises?
Though the deprecated `init()` function would cause a problem.

```
extension Promise where T == Void
{
    convenience init()
    {
        self.init(value: ())
    }
}
```

**Comment by mxcl** [comment]
Maybe this will work since Swift 3.1, since that syntax was only just added. PR welcome, if it works and all tests pass.

**Comment by RobinDaugherty** [comment]
For future searchers out there, the current way (Swift 4) to do create a fulfilled promise for a `Promise<Void>` is:

```swift
Promise<Void>.value(())
```

or for example:

```swift
doSomething().then { result -> Promise<Void> in
    if result.somethingIsIndicated {
        return doSomethingElse()
    } else {
        return Promise.value(())
    }
}
```

**Comment by mxcl** [comment]
@RobinDaugherty you can do `Promise()`.

This information is in our documentation.

**Comment by RobinDaugherty** [comment]
Thanks @mxcl that does work, but I don't think the code is as readable. On the other hand, I can't find this in the documentation, can you point me at that?

**Comment by ospfranco** [comment]
I'm trying this:

`return Promise<Void>.value(())` 

on Swift 5 and getting the following error: `Cannot call value of non-function type 'Promise<Void>'`, am I doing something wrong?

**Comment by dhoerl** [comment]
I'm converting some Objective-C code to Swift, and I have a method that returns a Promise<Void) using a Resolver:

func foo() -> Promise<Void>
    let p = Promise<Void>(resolver: { resolve in ...

Just to get this to compile, I need a proper "fulfill" argument. Lets say we define a constant to use:

  let voidValue : Void = ...
  resolver.fulfill(voidValue)

I found this above: `{ return }()`, and it compiles just fine. I then tried just `()` and that compiles too. However these do not:
  `Promise()`
  `Promise.void`
  `Promise.value(())`
 
Posting this to hopefully help some other dev who hits the same issue.
